### PR TITLE
Allow the user to set and get global api-root path

### PR DESCRIPTION
**Pull request by harenson**
Also fix sync repo.

### CODE REVIEW

File: test/trails_SUITE.erl [hunk 1]
```diff
@@ -291,6 +305,15 @@ trails_store(_Config) ->
   notfound = trails:retrieve("/other"),
   {comment, ""}.
 
+-spec trails_api_root(config()) -> ok.
+trails_api_root(_Config) ->
+  ok = trails:api_root("/api"),
+  "/api" = trails:api_root(),
+  Routes = [trails:trail("/things", the_handler)],
+  [{'_',[],[{[<<"api">>, <<"things">>],[],the_handler,[]}]}] =
```
Comment (gadgetci) [inline test/trails_SUITE.erl]: According to **Elvis**:

> Missing space after "," on line 313

Comment (gadgetci) [inline test/trails_SUITE.erl]: According to **Elvis**:

> Missing space after "," on line 313

File: Makefile [hunk 2]
```diff
@@ -1,17 +1,13 @@
 PROJECT = trails
 
 DEPS = cowboy cowlib ranch
+TEST_DEPS = xref_runner cowboy
```
Comment (elbrujohalcon) [inline Makefile]: `cowboy` is already in `DEPS` and `TEST_DEPS` _includes_ `DEPS` implicitely

File: src/trails.erl [hunk 3]
```diff
@@ -182,6 +184,16 @@ retrieve(Path) ->
       throw({not_started, trails})
   end.
 
+%% @doc Get api_root env param value if any, empty otherwise.
+-spec api_root() -> list().
```
Comment (elbrujohalcon) [inline src/trails.erl]: use `string()` instead of `list()`, it's clearer.

File: src/trails.erl [hunk 4]
```diff
@@ -182,6 +184,16 @@ retrieve(Path) ->
...
+api_root() ->
+    application:get_env(trails, api_root, "").
+
+%% @doc Set api_root env param to the given Path.
+-spec api_root(list()) -> ok.
```
Comment (elbrujohalcon) [inline src/trails.erl]: use `string()` instead of `list()`, it's clearer

**Pull request merged**